### PR TITLE
Separating edit and new partial for the sign up modal

### LINKED ISSUE #4173: Create signup modal
As an extension of #4133 , and similar to the issue of #2866 , we will need to create a modal that pops up when the user clicks the "signup" button in the navbar:
image

A modal should pop up, with the contents of this page: https://publiclab.org/signup
Here's the modal for logging in: 
image

I've made the login modal so if I'm not working on this task you can ask me for help :)

### PR DESCRIPTION
**Pull request by SidharthBansal**
Fixes #4173(<=== Add issue number here)

### CODE REVIEW

**Comment by SidharthBansal** [comment]
@JonathanXu1 good new is

**Comment by SidharthBansal** [comment]
@JonathanXu1 I rectified the error.
Now I guess everything is fine.
@sagarpreet-chadha can you please review this?
@oorjitchowdhary it will be great if you can 
* [ ] Sign Up form validations are working fine for incorrect input fields(testing)
* [ ] Sign Up modal validations are working fine for incorrect input fields(testing)
* [ ] Sign Up form signs a user in successfully(testing)
* [ ] Sign Up modal signs in a user successfully(testing)
* [ ] Sign Up works well on all the webpages(testing)
* [ ] Sign Up is redirected to the same page after user signs up(return_to)(testing)
* [ ] Check on the localhost(testing) for all above scenarios discussed.
* [ ] Check on the unstable branch(testing) for all above scenarios discussed.
* [ ] Check functionality of Oauth providers on the unstable branch.(testing)
So that I can reward you.
@JonathanXu1 ones @oorjitchowdhary tests it. You can close your pr and create one pr for the integration of login and signup modal.
Don't forget to push the changes of this pr into your integration pr @JonathanXu1. 
Please claim the task on the GCI dashboard so that @JonathanXu1 I can give you rewards.

**Comment by oorjitchowdhary** [comment]
@SidharthBansal Yes.. I'll help out.. Are the changes pushed to unstable?

**Comment by SidharthBansal** [comment]
Yes. You are helping this out from Jonathan's PR. It is continuation of
that PR.
He was stuck. So, just helping him out.

**Comment by oorjitchowdhary** [comment]
> * [ ]  Sign Up form validations are working fine for incorrect input fields(testing)
> * [ ]  Sign Up modal validations are working fine for incorrect input fields(testing)
> * [ ]  Sign Up form signs a user in successfully(testing)
> * [ ]  Sign Up modal signs in a user successfully(testing)
> * [ ]  Sign Up works well on all the webpages(testing)
> * [ ]  Sign Up is redirected to the same page after user signs up(return_to)(testing)
> * [ ]  Check on the localhost(testing) for all above scenarios discussed.
> * [ ]  Check on the unstable branch(testing) for all above scenarios discussed.
> * [ ]  Check functionality of Oauth providers on the unstable branch.(testing)
> * [ ]  return to works properly
>   So that I can reward you.

I'm on it

**Comment by SidharthBansal** [comment]
Great!!!

**Comment by oorjitchowdhary** [comment]
1. Sign Up form validations are working fine for incorrect input fields(testing)
2. Sign Up modal validations are working fine for incorrect input fields(testing)

If we put incorrect passwords in `password field` and `confirmation password` field, it shows this error..
Also, if you miss filling any field, it shows the same error.. 
On both, the modal and form page too..

image

**Comment by oorjitchowdhary** [comment]
3. Sign Up form signs a user in successfully(testing)
4. Sign Up modal signs in a user successfully(testing)
Unable to login after entering every field correctly.. 
The same error is shown..

screenshot 77

Also, the sign up modal redirects to the sign up page to show this error..

**Comment by oorjitchowdhary** [comment]
* [ ] Sign Up form signs a user in successfully(testing)

signup page error

* [ ] Sign Up modal signs in a user successfully(testing)

signup modal error

**Comment by oorjitchowdhary** [comment]
**Not working**
* [ ] Sign Up is redirected to the same page after user signs up(return_to)(testing)

publiclab_questionsanda

**Comment by SidharthBansal** [comment]
I hope above things work for the login modal. Am I right?

**Comment by oorjitchowdhary** [comment]
> I hope above things work for the login modal. Am I right?

On the login modal at unstable, after you log in you're redirected to publiclab.org

**Comment by oorjitchowdhary** [comment]
* [ ] Check functionality of Oauth providers on the unstable branch.(testing)
I'm redirected to publiclab.org (where I'm still not logged in) after I try to sign up from oauth on unstable..

**Comment by SidharthBansal** [comment]
Check for others.

**Comment by SidharthBansal** [comment]
Thanks

**Comment by oorjitchowdhary** [comment]
ok

**Comment by oorjitchowdhary** [comment]
> * [ ]  Check functionality of Oauth providers on the unstable branch.(testing)
> 
> I'm redirected to publiclab.org (where I'm still not logged in) after I try to sign up from oauth on unstable..

Wait a min.. I'm logged in at unstable.. 
So after using oauth signup, I'm redirected to publiclab.org but I'm logged in at unstable
Weird!

**Comment by oorjitchowdhary** [comment]
Unable to use unstable now.. I think you pushed some changes.. Build is going on currently

**Comment by SidharthBansal** [comment]
Actually some routes are set up for the public lab main production branch.
Specially the OAuth one. So, if you are logged in the unstable branch it's
great.
You can check the feature on the localhost too.

**Comment by SidharthBansal** [comment]
Yeah, I am sending the return_to changes.
In the meantime you can do the checking of login modal on the localhost.

**Comment by oorjitchowdhary** [comment]
> Actually some routes are set up for the public lab main production branch. Specially the OAuth one. So, if you are logged in the unstable branch it's great. 
Okay, that's wonderful..
> You can check the feature on the localhost too.

Yes.. I'll do that

**Comment by oorjitchowdhary** [comment]
Is the sign up modal merged? I can't access that from localhost.. I pulled the changes too

**Comment by SidharthBansal** [comment]
@oorjitchowdhary I am pushing some changes at unstable and in this pr. It will be great if in the meantime you can work on the other issues.

**Comment by oorjitchowdhary** [comment]
## Checks on Unstable
* [ ]  **ERROR** Sign Up form validations are working fine for incorrect input fields(testing) 
* [ ]  **ERROR** Sign Up modal validations are working fine for incorrect input fields(testing)

If we put incorrect passwords in password field and confirmation password field, it shows this error..
Also, if you miss filling any field, it shows the same error..
On both, the modal and form page too..

screenshot 77


* [ ]  **ERROR** Sign Up form signs a user in successfully(testing)

signup page error

Unable to login after entering every field correctly..
The same error is shown..

* [ ]  **ERROR** Sign Up modal signs in a user successfully(testing)

signup modal error

Unable to login after entering every field correctly..
The same error is shown..

* [ ]  **ERROR** Sign Up works well on all the webpages(testing)

* [ ]  Sign Up is redirected to the same page after user signs up(return_to)(testing)

Not working

* [ ]  Check functionality of Oauth providers on the unstable branch.(testing)

We're redirected publiclab.org but we're signed in through Oauth on unstable.publiclab.org

* [ ]  return to works properly

Not working

## Localhost
* [ ] **WORKING** Sign Up form validations are working fine for incorrect input fields(testing)

screenshot 78

* [ ] Sign Up modal validations are working fine for incorrect input fields(testing)
Redirected to sign up page for showing the error.

screenshot 78

* [ ] **WORKING** Sign Up form signs a user in successfully(testing)

screenshot 79

* [ ] **WORKING** Sign Up modal signs in a user successfully(testing)

screenshot 79

* [ ] **WORKING** Sign Up works well on all the webpages(testing)

screenshot 79

* [ ] **ERROR** Sign Up is redirected to the same page after user signs up(return_to)(testing)
Return_to not working

File: app/views/users/_form2.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div style="padding:10px;">
-<%= form_for :user, :as => :user, :url => :register + "?return_to=" + request.fullpath, :html => {:class => "row"} do |f| %>
+<%= form_for :user, :as => :user, :url => register_url + "?return_to=" + request.fullpath, :html => {:class => "row"} do |f| %>
```
Comment (SidharthBansal) [inline app/views/users/_form2.html.erb]: @oorjitchowdhary verify that this line is same. I have recently pushed the changes.

**Comment by SidharthBansal** [comment]
What about these for login modal?

**Comment by SidharthBansal** [comment]
Yeah, you can copy the changes and start debugging

**Comment by SidharthBansal** [comment]
@jonathanxu1 please claim the signup modal and other tasks upon which you worked to get rewarded

File: app/views/users/_form2.html.erb (hunk 1 above)
Comment (oorjitchowdhary) [reply app/views/users/_form2.html.erb]: I've to change this.. let me see again then

**Comment by JonathanXu1** [comment]
Yep, I just can't find the signup task:
image
Also should I claim the navbar icon task as well? @jywarren said my work would suffice for a task, but mine expired by then.

**Comment by JonathanXu1** [comment]
- [ ] Sign Up form validations are working fine for incorrect input fields(testing)
Page refreshes, modal closes
- [ ] Sign Up modal validations are working fine for incorrect input fields(testing)
Page refreshes, modal closes
- [ ] Sign Up form signs a user in successfully(testing)
Page refreshes, modal closes
- [x] Sign Up modal signs in a user successfully(testing)
Should work, but unable to verify
- [x] Sign Up works well on all the webpages(testing)
Should work, but unable to verify
- [x] Sign Up is redirected to the same page after user signs up(return_to)(testing)
Url matches original directory
- [ ] Check on the localhost(testing) for all above scenarios discussed.
@SidharthBansal is this the check for the master branch?
- [x] Check on the unstable branch(testing) for all above scenarios discussed.
1,2,3 ERROR
4,5,6 should work but redirects to https://localhost/register?return_to=/ instead of c9 link
- [x] Check functionality of Oauth providers on the unstable branch.(testing)
image
400 error on both modals and respective signup and login pages
- [x] return to works properly

**Comment by oorjitchowdhary** [comment]
> @oorjitchowdhary please test and give me the output for the login modal so that you can claim the same task and get rewards.

Okay.. Login modal checks on unstable and localhost both?

**Comment by SidharthBansal** [comment]
yeah

**Comment by oorjitchowdhary** [comment]
Okay..

**Comment by kevinzluo** [comment]
Hi there!

I will try these changes to help out too.

**Comment by SidharthBansal** [comment]
@kevinzluo great to hear that. I think testing is done.
It will be great if you can help us with the issues which have come.

**Comment by kevinzluo** [comment]
Currently I am facing the same problem with the signup modal as the login one.

I cannot click the password fields. Issue only occurs on firefox.

**Comment by SidharthBansal** [comment]
@JonathanXu1 testing is done by @oorjitchowdhary in these prs. You can focus on the coding stuff. Lets divide task equally.

**Comment by JonathanXu1** [comment]
Sounds good. I'll link the two modals for now.

**Comment by SidharthBansal** [comment]
@JonathanXu1 404 error is for google.
Can you try other providers?
I hope other providers are working fine.

**Comment by JonathanXu1** [comment]
@SidharthBansal I've created a new PR from this branch. Can I start work on the integration there? #4184

**Comment by SidharthBansal** [comment]
Thanks @oorjitchowdhary please claim the task and get rewards.

**Comment by oorjitchowdhary** [comment]
> Thanks @oorjitchowdhary please claim the task and get rewards.

I've claimed the sign up task

**Comment by SidharthBansal** [comment]
Moved to #4209

**Pull request closed without merging**